scripts/ci/processCoverageReport.py

```diff
@@ -0,0 +1,414 @@
+        return {'overall': {}, 'files': {}}
+
+def get_git_changed_files():
+    """Get changed files using git commands."""
+    try:
+        print("Getting changed files using git...")
+
+        base_branch = get_env_variable('GITHUB_BASE_REF', None)
+        pr_number = get_env_variable('PR_NUMBER', None)
+
+        if pr_number and base_branch:
+            print(f"PR detected (#{pr_number}). Getting files changed between {base_branch} and current branch")
+            cmd = ['git', 'diff', '--name-only', f'origin/{base_branch}...HEAD']
+        else:
+            print("Push event: Getting files changed in the last commit")
+            cmd = ['git', 'diff', '--name-only', 'HEAD^', 'HEAD']
+
+        print(f"Running command: {' '.join(cmd)}")
+        result = subprocess.run(cmd, capture_output=True, text=True)
+
+        if result.returncode != 0:
+            print(f"Error getting changed files with first approach: {result.stderr}")
+
+            cmd = ['git', 'diff', '--name-only', 'HEAD~10', 'HEAD']
+            print(f"Attempting alternative approach: {' '.join(cmd)}")
+            result = subprocess.run(cmd, capture_output=True, text=True)
+
+            if result.returncode != 0:
+                print(f"Error getting changed files with second approach: {result.stderr}")
+                return []
+
+        changed_files = [file for file in result.stdout.splitlines() if file.endswith('.java')]
+
+        print(f"Found {len(changed_files)} changed Java files")
+        if changed_files:
+            print("Changed Java files:")
+            for file in changed_files[:10]:
+                print(f"  - {file}")
+
+            if len(changed_files) > 10:
+                print(f"  ... and {len(changed_files) - 10} more files")
+
+        return changed_files
+    except Exception as e:
+        print(f"ERROR: Failed to get changed files: {str(e)}")
+        return []
+
+def match_changed_files_to_coverage(changed_files, coverage_data):
+    """Match changed files to coverage data."""
+    print("Matching changed files to coverage data...")
+    matches = []
+    unmatched = []
+
+    changed_file_names = [(f, os.path.basename(f).lower()) for f in changed_files]
+    print(f"Processing {len(changed_files)} changed files against {len(coverage_data)} coverage reports")
+
+    for changed_file, short_name in changed_file_names:
+        found = False
+
+        for report_index, report_coverage in enumerate(coverage_data):
+            files_coverage = report_coverage.get('files', {})
+
+            for coverage_path, coverage_info in files_coverage.items():
+                if changed_file.replace('\\', '/') in coverage_path:
+                    matches.append((changed_file, coverage_info))
+                    found = True
+                    print(f"Match found for {changed_file} in report #{report_index+1}")
+                    break
+
+                if short_name == os.path.basename(coverage_path).lower():
```

Comment:
I'm not sure I get this bit. Will this match files regardless of their location?